### ISSUE TITLE
Churn computation sometimes hangs/deadlocks

### ISSUE THREAD

**Issue by Nettifani** [o2sh/onefetch#1315]
### Duplicates

- [X] I have searched the existing issues

### Current behavior 😯

Sometimes onefetch hangs when computing churn information.

### Expected behavior 🤔

There should be no hang.

### Steps to reproduce 🕹

1. `cd` into a non-empty git repo.
2. Invoke `onefetch` repeatedly until the hang appears.

I can reproduce this issue pretty reliably on my computer with different repositories.

### Additional context/Screenshots 🔦

Operating system: Arch Linux
Version of onefetch being used: `onefetch 2.20.0` (from the Arch repositories)

I have attached a pretty big stack trace from GDB.
[onefetch-trace.log](https://github.com/o2sh/onefetch/files/15075418/onefetch-trace.log)

### Possible Solution 💡

I'm guessing there's a race condition here (which then causes a deadlock):
https://github.com/o2sh/onefetch/blob/69a0609e9da1a6a080d6b6ce1f7ef3f08992b1a5/src/info/git/mod.rs#L185-L197 https://github.com/o2sh/onefetch/blob/69a0609e9da1a6a080d6b6ce1f7ef3f08992b1a5/src/info/git/mod.rs#L105-L110
which would sometimes go like this:
 1. Thread A sends a commit id to the churn thread.
 2. The churn thread processes that commit id and then checks whether it should stop by atomically checking `has_commit_graph_traversal_ended` and `total_number_of_commits`.
 3. Thread A sets `has_commit_graph_traversal_ended` and `total_number_of_commits`.
 4. However, the churn thread has already done the stop check and is now waiting for the next commit id to process.
 5. Thread A waits on the churn thread, whilst the churn thread waits on thread A.

One way to fix the deadlock is to add a `drop(churn_tx)` before joining the churn thread.
